application/controllers/artifacts_controller.py

```diff
@@ -69,11 +74,12 @@ def create(self, params):
         metadata = self._upload_file(params)
         artifact = self._create_artifact(params, metadata)
         ImageRecognizer.auto_add_tags(artifact.elastic)
-        return respond_with(artifact.elastic), 200
+        return respond_with(artifact.neo), 200
 
     def _create_artifact(self, params, metadata):
         params.update(metadata)
-        artifact = Artifact(**params)
+        artifact = ArtifactBuilder()
+        artifact.build_with(**params)
```

Comment:
I get that you want the ArtifactBuilder to be a builder pattern for the elastic and neo models, but having this and the rails like Model methods (like find_by etc.) on that Builder as well seems a bit unintuitive. 
So maybe don't call it builder or extract the builder part from the model wrapper part.

Reply:
Well, as I said the class is in a very rough state since it has at least two responsibilities right now. I think it will get much more clear once I extract all the ES stuff from it. Then all the find_by calls for example can also be directly executed on the Artifact and not the ArtifactBuilder. You are right though that the name is not very expressive right now. It's difficult to find one that explains all the responsibilites right now though...

Reply:
My overall idea is to:
- Have Artifacts be accessed directly for index/show routes
- Have the artifact builder be a small helper class for creating Artifacts (and maybe updating)
- Have a new class ElasticSearchSynchronizer, that manages populating elasticsearch data. And is called in a post_save hook on the Neo4J models.

I want to do all three in a seperate PR since this one is already a pain to read and very large :D